htdocs/webportal/class/webPortalTheme.class.php

```diff
@@ -0,0 +1,147 @@
+	);
+
+
+	public $loginLogoUrl;
+	public $menuLogoUrl;
+	public $loginBackground;
+
+
+	/**
+	 * Constructor
+	 */
+	public function __construct()
+	{
+		$this->loadPrimaryColor();
+
+		$this->loginLogoUrl = getDolGlobalString('WEBPORTAL_LOGIN_LOGO_URL');
+		$this->menuLogoUrl = getDolGlobalString('WEBPORTAL_MENU_LOGO_URL', $this->loginLogoUrl);
+		$this->loginBackground = getDolGlobalString('WEBPORTAL_LOGIN_BACKGROUND');
+		$this->bannerBackground = getDolGlobalString('WEBPORTAL_BANNER_BACKGROUND');
+		$this->bannerUseDarkTheme = getDolGlobalInt('WEBPORTAL_BANNER_BACKGROUND_IS_DARK');
+	}
+
+	/**
+	 * return hex of primary theme color
+	 *
+	 * @return string
+	 */
+	public function loadPrimaryColor()
+	{
+		global $conf;
+
+		$outColor = '';
+
+		if (!empty($conf->global->WEBPORTAL_PRIMARY_COLOR)) {
+			$outColor = $conf->global->WEBPORTAL_PRIMARY_COLOR;
+		} elseif (!empty($conf->global->THEME_ELDY_TOPMENU_BACK1)) {
+			$outColor = '#' . colorArrayToHex(colorStringToArray($conf->global->THEME_ELDY_TOPMENU_BACK1));
+		}
+
+		if (empty($outColor) || !colorValidateHex($outColor)) {
+			$outColor = '#263c5c';
+		}
+
+		$this->primaryColorHex = $outColor;
+		$this->primaryColorHsl = $this->colorHexToHsl($outColor, true, true);
+	}
+
+	/**
+	 * @param string 	$hex 			color in hex
+	 * @param float|false 	$alpha 			0 to 1 to add alpha channel
+	 * @param bool 		$returnArray	true=return an array instead, false=return string
+	 * @return string|array				String or array
+	 */
+	public function colorHexToHsl($hex, $alpha = false, $returnArray = false)
+	{
+		if (function_exists('colorHexToHsl')) {
+			return colorHexToHsl($hex, $alpha, $returnArray);
+		}
+
+		// For retro compatibility
+		// TODO : Remove this when webportal is included in DOLIBARR standard
+		$hex      = str_replace('#', '', $hex);
+		$red = hexdec(substr($hex, 0, 2)) / 255;
+		$green = hexdec(substr($hex, 2, 2)) / 255;
+		$blue = hexdec(substr($hex, 4, 2)) / 255;
+
+		$cmin = min($red, $green, $blue);
+		$cmax = max($red, $green, $blue);
+		$delta = $cmax - $cmin;
+
+		if ($delta == 0) {
+			$hue = 0;
+		} elseif ($cmax === $red) {
+			$hue = (($green - $blue) / $delta);
+		} elseif ($cmax === $green) {
+			$hue = ($blue - $red) / $delta + 2;
+		} else {
+			$hue = ($red - $green) / $delta + 4;
+		}
+
+		$hue = round($hue * 60);
+		if ($hue < 0) {
+			$hue += 360;
+		}
+
+		$lightness = (($cmax + $cmin) / 2);
+		$saturation = $delta === 0 ? 0 : ($delta / (1 - abs(2 * $lightness - 1)));
+		if ($saturation < 0) {
+			$saturation += 1;
+		}
+
+		$lightness = round($lightness*100);
+		$saturation = round($saturation*100);
+
+		if ($returnArray) {
+			return array(
+				'h' => $hue,
+				'l' => $lightness,
+				's' => $saturation,
+				'a' => $alpha === false ? 1 : $alpha
+			);
+		} elseif ($alpha) {
+			return 'hsla(' . $hue . ', ' . $saturation . ', ' . $lightness . ' / ' . $alpha . ')';
+		} else {
+			return 'hsl(' . $hue . ', ' . $saturation . ', ' . $lightness . ')';
+		}
+	}
```

Comment:
```suggestion
		if (!function_exists('colorHexToHsl')) {
				require_once DOL_DOCUMENT_ROOT . '/core/lib/functions2.lib.php';
		}

			return colorHexToHsl($hex, $alpha, $returnArray);
	}
```

Reply:
le top serait de remplacer les $this->colorHexToHsl par colorHexToHsl mais il faut bien que lib soit chargée